src/Microsoft.ML.Tokenizers/PACKAGE.md

```diff
@@ -0,0 +1,57 @@
+## About
+
+Microsoft.ML.Tokenizers supports various the implmentation of the tokenization used in the NLP transforms.
+
+## Key Features
+
+* Extensisble tokenizer architecture that allows for specialization of Normalizer, PreTokenizer, Model/Encoder, Decoder
+* BPE - Byte pair encoding model
+* English Roberta model
+* Tiktoken model
+
+## How to Use
+
+```c#
+using Microsoft.ML.Tokenizers;
+
+// initialize the tokenizer for `gpt-4` model, downloading data files
+Tokenizer tokenizer = await Tiktoken.CreateByModelNameAsync("gpt-4");
+
+string source = "Text tokenization is the process of splitting a string into a list of tokens.";
+
+Console.WriteLine($"Tokens: {tokenizer.CountTokens(source)}");
+// print: Tokens: 16
+
+var trimIndex = tokenizer.LastIndexOfTokenCount(source, 5, out string processedText, out _);
+Console.WriteLine($"5 tokens from end: {processedText.Substring(trimIndex)}");
+// 5 tokens from end:  a list of tokens.
```

Comment:
Does this look correct, @tarekgh?  I asked for 5 tokens and got this (which looks like 4), same for the prefix case.

Reply:
Yes it is correct :-)

![image](https://github.com/dotnet/machinelearning/assets/10833894/4473919b-fdde-40aa-b2c6-e9b6ad4072dc)

Reply:
Interesting.  Cool tool.

Reply:
https://platform.openai.com/tokenizer